backend/src/app.js

```diff
@@ -61,6 +61,9 @@ app.use(function (req, res, next) {
   if (!req.session.gptModel) {
     req.session.gptModel = defaultModel;
   }
+  if (!req.session.phases) {
+    req.session.phases = getPhases();
+  }
```

Comment:
As discussed we should only need to store one variable here, I'm thinking calling it something like `numPhasesCompleted` and initialising it to `0`. Then we can go up to `3`, indicating that all three phases have been completed by the user.